src/Evolution/Systems/GeneralizedHarmonic/DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,396 @@
+    r_width);
+  auto mu_L1 = roll_on * get(weight) * pow<4>(log_sqrtg_over_lapse);
+  auto mu_L2 = roll_on * get(weight) * pow<4>(log_one_over_lapse);
+  auto mu_S = mu_L1;
+  auto h_prefac1 = mu_L1 * log_sqrtg_over_lapse + mu_L2 * log_one_over_lapse;
+  auto h_prefac2 = -mu_S / get(lapse);
+  // Calculate H_a
+  auto gauge_h =
+    make_with_value<typename GaugeH<SpatialDim, Frame>::type>(get(lapse), 0.);
+  for (size_t a = 0; a < SpatialDim+1; ++a) {
+    gauge_h.get(a) = (1. - roll_on) * GaugeH_init.get(a)
+                   + h_prefac1 * spacetime_unit_normal_one_form.get(a);
+    for (size_t i = 0; i < SpatialDim; ++i) {
+      gauge_h.get(a) += h_prefac2 * spacetime_metric.get(a, i+1) * shift.get(i);
+    }
+  }
+  return gauge_h;
+}
+
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Spacetime derivatives of the damped Harmonic gauge source function.
+ *
+ * \details Compute spacetime derivatives of the damped harmonic gauge source
+ *  function, i.e. \f$\partial_a H_b\f$, where the source function H is
+ * \f{align*}
+ * H_a := T1 + T2 + T3
+ * \f}
+ * with:
+ * \f$ T1 = H_a^\mathrm{init}[1 - R(t)] \f$,
+ * \f$ T2 = [\mu_{L1} log(\sqrt{g}/N) + \mu_{L2} log(1/N)] n_a \f$,
+ * \f$ T3 =  - \mu_S g_{ai} N^i / N \f$.
+ *
+ * See the header file for \f$\partial_a T1,2,3 \f$.
+ * */
+template <size_t SpatialDim, typename Frame>
+typename SpacetimeDerivGaugeH<SpatialDim, Frame>::type
+SpacetimeDerivDampedHarmonicHCompute<SpatialDim, Frame>::function(
+    const typename InitialGaugeH<SpatialDim, Frame>::type& GaugeH_init,
+    const typename SpacetimeDerivInitialGaugeH<SpatialDim, Frame>::type&
+      dGaugeH_init,
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, SpatialDim, Frame>& shift,
+    const tnsr::A<DataVector, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::a<DataVector, SpatialDim, Frame>&
+      spacetime_unit_normal_one_form,
+    const Scalar<DataVector>& sqrt_det_spatial_metric,
+    const tnsr::II<DataVector, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::aa<DataVector, SpatialDim, Frame>& spacetime_metric,
+    const tnsr::iaa<DataVector, SpatialDim, Frame>& d_spacetime_metric,
+    const tnsr::aa<DataVector, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataVector, SpatialDim, Frame>& phi,
+    const Scalar<DataVector>& gamma1,
+    const double& time, const double& t_start, const double& sigma_t,
+    const tnsr::I<DataVector, SpatialDim, Frame>& coords,
+    const double& r_max, const double& r_width) noexcept {
+  // 3+1 quantities
+  auto spatial_metric = gr::spatial_metric<SpatialDim, Frame, DataVector>(
+    spacetime_metric);
+  auto det_spatial_metric =
+    get(sqrt_det_spatial_metric) * get(sqrt_det_spatial_metric);
+  auto inverse_spacetime_metric =
+    gr::inverse_spacetime_metric<SpatialDim, Frame, DataVector>(lapse, shift,
+      inverse_spatial_metric);
+  auto d0_spatial_metric = time_deriv_of_spatial_metric(lapse, shift,
```

Comment:
In the end we have to use an expression for `d_spacetime_metric` that is free of numerical derivatives, i.e. `d_spacetime_metric` will be expressed in terms of the reduction variable phi. We can assume this already here to optimize `time_deriv_of_spatial_metric`.
The same is true for `deriv_spatial_metric`.

Reply:
I'm not sure I understand this comment vis-a-vis the corresponding lines of code that have been highlighted. `GeneralizedHarmonic::deriv_spatial_metric` already uses `phi`. Could you please rephrase what you mean here?

Reply:
In the GHG system the derivatives of H must be expressed without using any numerical derivatives of the fields, i.e. you must use the reduction variables `phi` and `pi`. Thus the use of `d_spacetime_metric` in this function would be redundant, since it will also be expressed using `phi` and `pi`.
For the computation of in `time_deriv_of_spatial_metric` you are using the GHG system right hand side, but it would be more efficient reconstruct d0 from the reduction constraints directly.

You are right regarding `GeneralizedHarmonic::deriv_spatial_metric`, I don't know what I was thinking.

Reply:
Yep, the assumption in `time_deriv_of_spatial_metric` is that its input `d_spacetime_metric` has been computed from pi and phi and time derivs of lower shift etc, but I see your point that we can avoid the necessity of `d_spacetime_metric` while computing damped wave gauge function's derivs if we changed the function `time_deriv_of_spatial_metric` to assemble derivatives of spacetime metric in-house from `pi, phi` and other first-order vars.

Since `time_deriv_of_spatial_metric` is part of another PR (#1179 ) I will address this point there first. Once that  PR is merged, I'll rebase this PR and remove the need for `d_spacetime_metric` iin computing gauge function H and its derivs.

Reply:
I would prefer that there be no function `time_deriv_of_spatial_metric` at all.  That function seems to be used only once right here, and once in the shift.  Eliminating that function (which is very simple and inexpensive) in favor of computing the correct thing inside of the code that needs it, would eliminate the possibility of passing in the wrong (i.e. non-hyperbolic) thing. See my comments on #1177 and #1179.

Reply:
This discussion ties closely together with those on [PR #1179 ](https://github.com/sxs-collaboration/spectre/pull/1179#discussion_r239981773) and on [PR #1177 ](https://github.com/sxs-collaboration/spectre/pull/1177#discussion_r239984144), and has primarily been brought on because my (now old) code for `time_deriv_of_spatial_metric` function was accepting `d_spacetime_metric` as an argument. Sorry for all the confusion this has brought on.  I have already changed `time_deriv_of_spatial_metric` in #1179 (where it originates) to have it assemble its output from `phi` and `pi` **only**, and now I've updated that function in this PR as well.

I would still argue for keeping `time_deriv_of_spatial_metric` as a function, as it is used at least two places -- once here and once (should be) in `time_deriv_of_lower_shift` in #1177 .

@markscheel :
> I would be ok with this function existing only if  it took as arguments `pi`, `phi`, `lapse`, and `shift` (and nothing else)

It does now. Everywhere.

>  and if the function name were something like extract_dt_spatial_metric_from_pi_and_phi that makes clearer what it is.

Changing this function's name would break the naming convention in ComputeGhQuantities.cpp as most functions there are indeed extracting various derivs from `pi` and `phi` and then would need to be renamed `extract_something_from_pi_and_phi` as well. Eg. `deriv_spatial_metric` which is trivially extracted from `phi` (see [here](https://github.com/sxs-collaboration/spectre/blob/develop/src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp#L154) ). Now that the arguments of `time_deriv_of_spatial_metric` are more obvious, i.e. only `pi`, `phi`, `lapse`, and `shift`, hopefully it'd be transparent to the user where is it getting its derivatives from (plus the fact that all other functions in its file extract derivs from GH vars too).

Reply:
ping @markscheel